overlord/snapstate/snapstate.go

```diff
@@ -74,6 +74,87 @@ var ErrNothingToDo = errors.New("nothing to do")
 
 var osutilCheckFreeSpace = osutil.CheckFreeSpace
 
+type minimalInstallInfo interface {
+	InstanceName() string
+	Type() snap.Type
+	SnapBase() string
+	DownloadSize() int64
+	Prereq(st *state.State) []string
+}
+
+type updateParamsFunc func(*snap.Info) (*RevisionOptions, Flags, *SnapState)
+
+type readyUpdateInfo interface {
+	minimalInstallInfo
+
+	SnapSetupForUpdate(st *state.State, params updateParamsFunc, userID int, globalFlags *Flags) (*SnapSetup, *SnapState, error)
```

Comment:
Not sure about `SnapSetup..`, we get snapstate too (although maybe that's implied by the fact there's Update in the name). Maybe SnapUpdateDataForParams?

Reply:
I named this before I noticed I had to return SnapState too, OTOH I expect we'll want to refactor this again soon (because in the new world params itself is a bit awkward), so I think this name is good enough for now.